### DISCUSSION TITLE
I want to get an iModel ID for an iTwin

### DISCUSSION THREAD

**Discussion by hussainshahzad5471** [iTwin/community#46] [Q&A] [answered]
I want to get the iModel ID for an iTwin, didn't get anything in the documentation, then i looked at the har file to see how developer.bentley.com does it. looks like they hit the api "https://api.bentley.com/imodels/?iTwinId={}&$orderBy=name%20asc&$skip=0&$top=100" , i tried hitting the same from postman but i get error :
{
    "error": {
        "code": "InvalidiModelsRequest",
        "message": "Cannot get iModels.",
        "details": [
            {
                "code": "MissingRequiredParameter",
                "message": "Required parameter is missing.",
                "target": "projectId"
            }
        ]
    }
}
then i passed the project ID (got from a parameter vtp_projectId from the har file) and i get a new error:
{
    "error": {
        "code": "ProjectNotFound", 
        "message": "Requested project is not available."
    }
}

**Comment by GytisCepk** [comment, marked as answer]
Try getting iModel ID for your iTwin in [iModels API documentation page](https://developer.bentley.com/apis/imodels-v2/operations/get-itwin-imodels/):

![image](https://github.com/iTwin/community/assets/98940208/a6ca6002-a9ea-49e7-a22a-f1956c1dba3a)

If you encounter same issues there, let us know.

**Reply by hussainshahzad5471** [reply]
<img width="634" alt="image" src="https://github.com/iTwin/community/assets/167759662/07a39413-d23b-48b2-b3f7-e50e7f45cf98">
it works from https://developer.bentley.com/apis/imodels-v2/operations/get-itwin-imodels/ but not from postman or from my react application, please see the screenshot.

**Reply by hussainshahzad5471** [reply]
<img width="574" alt="image" src="https://github.com/iTwin/community/assets/167759662/a984c004-0ac7-440a-8b97-683313de4632">

**Reply by GytisCepk** [reply]
It looks like you're calling an old version of iModels API. 

Try adding `Accept: application/vnd.bentley.itwin-platform.v2+json` header to any requests made to iModels API.

**Comment by hussainshahzad5471** [comment]
@GytisCepk, thanks a ton. that works, do I have to use this header in all my requests or only the ones not working ?

**Reply by GytisCepk** [reply]
Only in requests which call iModels API (https://api.bentley.com/imodels/). For other APIs (e.g. iTwins API - https://api.bentley.com/itwins/) you should not pass it.
